packages/dx-chart-core/src/utils/animation.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  EasingFn, SetAttributeFn, AnimationFn, GetNewPositionsFn,
+  BarCoordinates, PointCoordinates,
+  PathCoordinates, PieCoordinates, PathPoints,
+} from '../types';
+
+/** @internal */
+const getProgress = ({ elapsed, total }: {elapsed: number, total: number}) =>
+Math.min(elapsed / total, 1);
+
+/** @internal */
+const runAnimation = (
+  setAttributes: SetAttributeFn,
+  getNewPositions: GetNewPositionsFn,
+  easing: EasingFn, duration: number, delay: number,
+) => {
+  const promise = () => new Promise((resolve) => {
+    setTimeout(() => {
+      const time = {
+        start: new Date().getTime(),
+        total: duration,
+        elapsed: 0,
+      };
+      const step = () => {
+        time.elapsed = new Date().getTime() - time.start;
+        const progress = getProgress(time);
+        setAttributes(getNewPositions(easing(progress)));
+
+        if (progress < 1) requestAnimationFrame(step);
+      };
+      resolve(requestAnimationFrame(step));
+    }, delay);
+  });
+
+  return promise().then((res) => {
+    return res;
+  });
```

Comment:
I think we should move the `then` part into a function when we use `runAnimation` because of the `runAnumation` function now returns a Promise instead of Number.